comp/core/workloadmeta/store.go

```diff
@@ -701,13 +705,20 @@ func (w *workloadmeta) handleEvents(evs []CollectorEvent) {
 	// process an event.
 	w.storeMut.Unlock()
 
+	var wg sync.WaitGroup
+	wg.Add(len(filteredEvents))
 	for sub, evs := range filteredEvents {
 		if len(evs) == 0 {
+			wg.Done()
 			continue
 		}
 
-		w.notifyChannel(sub.name, sub.ch, evs, true)
+		go func(sub subscriber, evs []Event) {
+			w.notifyChannel(sub.name, sub.ch, evs, true)
+			wg.Done()
+		}(sub, evs)
```

Comment:
Before:
`handleEvents()` waits at most `len(filteredEvents) * eventBundleChTimeout`. *eventBundleChTimeout = 1 second.

After
`handleEvents()` waits at most `eventBundleChTimeout`.

Question:
Anyway, why should we wait `subscriber` function completion?
Is it just for waning logs?

https://github.com/DataDog/datadog-agent/blob/294ae46153c39545b7e9517a65fed146f50827fb/comp/core/workloadmeta/store.go#L772-L774